SpatialGDK/Source/SpatialGDKTest/Private/Tests/SpatialReceiverTest.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Misc/AutomationTest.h"
+#include "Interop/SpatialReceiver.h"
+
+namespace SpatialReceiverTestUtils {
+	bool SpatialReceiverOnEntityQueryResponseTest(Worker_EntityQueryResponseOp& TestOp)
+	{
+		USpatialReceiver* Receiver = NewObject<USpatialReceiver>();
+		Worker_RequestId RequestId = 0;
+		EntityQueryDelegate TestQueryDelegate;
+
+		bool IsDelegateCalled = false;
+
+		TestQueryDelegate.BindLambda([&IsDelegateCalled](Worker_EntityQueryResponseOp& _)
+		{
+			IsDelegateCalled = true;
+		});
+
+		Receiver->AddEntityQueryDelegate(RequestId, TestQueryDelegate);
+
+		Receiver->OnEntityQueryResponse(TestOp);
+
+		return IsDelegateCalled;
+	}
+}
+
+IMPLEMENT_SIMPLE_AUTOMATION_TEST(FSpatialReceiverOnEntityQueryResponseTestStatusCodeSuccess, "SpatialGDK.USpatialReceiver.OnEntityQueryResponse.CallsDelegateIfStatusCodeIsSuccess", EAutomationTestFlags::EditorContext | EAutomationTestFlags::EngineFilter)
+
+bool FSpatialReceiverOnEntityQueryResponseTestStatusCodeSuccess::RunTest(const FString& Parameters)
+{
+	Worker_EntityQueryResponseOp TestQueryResponseOp{};
+	TestQueryResponseOp.status_code = Worker_StatusCode::WORKER_STATUS_CODE_SUCCESS;
+
+	return SpatialReceiverTestUtils::SpatialReceiverOnEntityQueryResponseTest(TestQueryResponseOp);
```

Comment:
What is the error message it this function fails?

Reply:
It shows red in the editor with a link to this line. A developer would navigate to this and understand the delegate isn't being called. They will then put some breakpoints and investigate why.

When I have some learning time, I'll try to make some improvements to the hot-reloading flow to make running tests while making changes more intuitive. Currently it's a bit of a pain so I wouldn't encourage people to write tests yet.